target/arc/semfunc.c

```diff
@@ -55,27 +55,52 @@ arc_gen_cmpl2_i64(TCGv_i64 ret, TCGv_i64 arg1,
 }
 
 void
+/**
+ * In a signed addition, there is an overflow when these conditions are met:
+ * (1) Both operands are of the same sign.
+ * (2) The result's sign is different from the operand's.
+ */
+arc_gen_add_signed_overflow_tl(TCGv overflow, TCGv result,
+                               TCGv op1, TCGv op2, uint8_t operand_size)
+{
+    TCGv diff_operand_sign = tcg_temp_new();
+    TCGv diff_result_sign = tcg_temp_new();
+
```

Comment:
there is a logical order for what I suggested earlier. else, a curious reader would ask why on earth `result` is getting xored with `op1` and not `op2`.
also don't remove the comment below.
```
tcg_gen_xor_tl(diff_operand_sign, op1, op2);
tcg_gen_xor_tl(diff_result_sign, op1, result);
/* if (diff_result_sign AND !diff_operand_sign) */
tcg_gen_andc(overflow, diff_result_sign, diff_operand_sign);
```